flink-end-to-end-tests/flink-tpcds-test/src/main/java/org/apache/flink/table/tpcds/utils/TpcdsResultComparator.java

```diff
@@ -61,7 +62,8 @@ public static void main(String[] args) throws Exception {
 
         for (String queryId : VALIDATE_QUERIES) {
             File expectedFile = new File(expectedDir, queryId + RESULT_SUFFIX);
-            File actualFile = new File(actualDir, queryId + RESULT_SUFFIX);
+            File actualFileDirectory = new File(actualDir, queryId + RESULT_SUFFIX);
+            File actualFile = Objects.requireNonNull(actualFileDirectory.listFiles())[0];
```

Comment:
Would you explain a bit that why the actual file is changed in this case?
Does this means that there is behavior change when switching from an old csv sink to new a new csv sink?

Reply:
the output of old csv sink is a file, but for new csv sink, output is a directory. so we need add a step to get the actualFile.